### PR TITLE
Side navigation enhancements

### PR DESCRIPTION
**Pull request by Andrew-K-Lam**
## Description:

Added the following features:
1. Animation on clicking submenu
2. Consistent spacing above and below the sidenav (verticalSpacing)
3. Added spacing above and below list of links in the submenu

## Checklist before submitting pull request
* Commits follow our [Developer Guide](https://github.com/telus/tds-community/blob/chore/template-hints/.github/CONTRIBUTING.md)
* [x]  New code is unit tested
* [x]  make sure visual and accessibility tests pass
* [x]  make sure code builds

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -121,6 +121,7 @@
     "react": "^16.3.2",
     "react-dom": "^16.3.2",
     "react-styleguidist": "^8.0.6",
+    "react-transition-group": "^2.2.1",
```
Comment (jraff) [inline package.json]: This package be added to the dependencies of SideNavigation.

Comment (Andrew-K-Lam) [reply package.json]: react-transition-group is used in the /shared directory and not in SideNavigation....

**Comment by invalidred** [comment]
@jraff is this still an issue?

**Comment by Andrew-K-Lam** [comment]
@invalidred This PR is still open and active.  This PR is awaiting action on this PR: https://github.com/telus/tds-core/pull/957

File: package.json (hunk 1 above)
Comment (Andrew-K-Lam) [reply package.json]: Removed from package.json

**Comment by donnavitan** [comment]
Adding notes for future reference:
* The dimple shadow looks different on the coded component. May need to update design/code, whichever is prefered.
* May need an active state for the submenu item.
* Latest artwork https://telus.invisionapp.com/share/7ER8FQXRHMD . Please note that when you inspect this, the padding will show as 15px top and 17px bottom. This accounts for the visual alignment for the text within Sketch. However, the correct padding in the coded component will be 16px top and 16px bottom.
* Latest Sketch file https://invis.io/a/YK5A9AYDDTR6
* Received Elena's design approval

**Pull request merged**